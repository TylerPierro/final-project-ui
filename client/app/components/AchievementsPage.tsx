import React, { Component } from 'react'
import { connect } from 'react-redux'

export class AchievementsPage extends Component {
  //@ts-ignore
  render = () => {
    return (
      <div>
        <h1>Achievements</h1>
      </div>
    )
  }
}

export default connect()(AchievementsPage)
