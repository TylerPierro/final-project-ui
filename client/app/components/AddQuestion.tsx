import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import fontawesome from '@fortawesome/fontawesome'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { startcreateNewQuestion } from '../actions/create'

export class AddQuestion extends Component {
  state = {
    newQuestions: [],
    currentQuestion: {
      author: 'dserna',
      answers: [
        { answer: '', percentPoints: 0, feedback: '' },
        { answer: '', percentPoints: 0, feedback: '' }
      ],
      format: 'true-false'
    },
    answerItem: {}
  }

  private updateTitle = (e: any) => {
    const title = e.target.value
    this.setState({
      currentQuestion: {
        ...this.state.currentQuestion,
        title
      }
    })
  }

  private updateFormat = (e: any) => {
    console.log(e.target.value)
    const format = e.target.value
    console.log('format ', format)
    if (format === 'true-false') {
      this.setState({
        currentQuestion: {
          ...this.state.currentQuestion,
          format: format,
          answers: [
            { answer: '', percentPoints: 0, feedback: '' },
            { answer: '', percentPoints: 0, feedback: '' }
          ]
        }
      })
    } else {
      this.setState({
        currentQuestion: {
          ...this.state.currentQuestion,
          format: format,
          answers: [
            { answer: '', percentPoints: 0, feedback: '' },
            { answer: '', percentPoints: 0, feedback: '' },
            { answer: '', percentPoints: 0, feedback: '' },
            { answer: '', percentPoints: 0, feedback: '' }
          ]
        }
      })
    }
    console.log('updateFormat Called')
    console.log(this.state)
  }

  private updateArr = (e: any, arg1: number, arg2: string) => {
    let newAnswersArr = this.state.currentQuestion.answers
    newAnswersArr[arg1][arg2] = e.target.value
    this.setState({
      currentQuestion: {
        ...this.state.currentQuestion,
        answers: newAnswersArr
      }
    })
    console.log('updateArr called')
    console.log(this.state)
  }

  private createTable = () => {
    switch (this.state.currentQuestion.format) {
      case 'true-false':
        return (
          <form>
            <div className="row">
              <label htmlFor="true-false-answer">Answer</label>
              <input
                type="text"
                id="true-false-answer"
                value={this.state.currentQuestion.answers[0].answer}
                onChange={(e: any) => {
                  this.updateArr(e, 0, 'answer')
                }}
              />
              <label htmlFor="true-false-percent-points">Percent Points</label>
              <input
                type="text"
                id="true-false-percent-points"
                value={this.state.currentQuestion.answers[0].percentPoints}
                onChange={(e: any) => {
                  this.updateArr(e, 0, 'percentPoints')
                }}
              />
              <label htmlFor="true-false-feed-back">feed Back</label>
              <input
                type="text"
                id="true-false-feed-back"
                value={this.state.currentQuestion.answers[0].feedback}
                onChange={(e: any) => {
                  this.updateArr(e, 0, 'feedback')
                }}
              />
            </div>
            <div className="row">
              <label htmlFor="true-false-answer">Answer</label>
              <input
                type="text"
                id="true-false-answer"
                value={this.state.currentQuestion.answers[1].answer}
                onChange={(e: any) => {
                  this.updateArr(e, 1, 'answer')
                }}
              />
              <label htmlFor="true-false-percent-points">Percent Points</label>
              <input
                type="text"
                id="true-false-percent-points"
                value={this.state.currentQuestion.answers[1].percentPoints}
                onChange={(e: any) => {
                  this.updateArr(e, 1, 'percentPoints')
                }}
              />
              <label htmlFor="true-false-feed-back">feed Back</label>
              <input
                type="text"
                id="true-false-feed-back"
                value={this.state.currentQuestion.answers[1].feedback}
                onChange={(e: any) => {
                  this.updateArr(e, 1, 'feedback')
                }}
              />
            </div>
          </form>
        )
      default:
        return (
          <form>
            <div className="row">
              <label htmlFor="true-false-answer">Answer</label>
              <input
                type="text"
                id="true-false-answer"
                value={this.state.currentQuestion.answers[0].answer}
                onChange={(e: any) => {
                  this.updateArr(e, 0, 'answer')
                }}
              />
              <label htmlFor="true-false-percent-points">Percent Points</label>
              <input
                type="text"
                id="true-false-percent-points"
                value={this.state.currentQuestion.answers[0].percentPoints}
                onChange={(e: any) => {
                  this.updateArr(e, 0, 'percentPoints')
                }}
              />
              <label htmlFor="true-false-feed-back">feed Back</label>
              <input
                type="text"
                id="true-false-feed-back"
                value={this.state.currentQuestion.answers[0].feedback}
                onChange={(e: any) => {
                  this.updateArr(e, 0, 'feedback')
                }}
              />
            </div>
            <div className="row">
              <label htmlFor="true-false-answer">Answer</label>
              <input
                type="text"
                id="true-false-answer"
                value={this.state.currentQuestion.answers[1].answer}
                onChange={(e: any) => {
                  this.updateArr(e, 1, 'answer')
                }}
              />
              <label htmlFor="true-false-percent-points">Percent Points</label>
              <input
                type="text"
                id="true-false-percent-points"
                value={this.state.currentQuestion.answers[1].percentPoints}
                onChange={(e: any) => {
                  this.updateArr(e, 1, 'percentPoints')
                }}
              />
              <label htmlFor="true-false-feed-back">feed Back</label>
              <input
                type="text"
                id="true-false-feed-back"
                value={this.state.currentQuestion.answers[1].feedback}
                onChange={(e: any) => {
                  this.updateArr(e, 1, 'feedback')
                }}
              />
            </div>
            <div className="row">
              <label htmlFor="true-false-answer">Answer</label>
              <input
                type="text"
                id="true-false-answer"
                value={this.state.currentQuestion.answers[2].answer}
                onChange={(e: any) => {
                  this.updateArr(e, 2, 'answer')
                }}
              />
              <label htmlFor="true-false-percent-points">Percent Points</label>
              <input
                type="text"
                id="true-false-percent-points"
                value={this.state.currentQuestion.answers[2].percentPoints}
                onChange={(e: any) => {
                  this.updateArr(e, 2, 'percentPoints')
                }}
              />
              <label htmlFor="true-false-feed-back">feed Back</label>
              <input
                type="text"
                id="true-false-feed-back"
                value={this.state.currentQuestion.answers[2].feedback}
                onChange={(e: any) => {
                  this.updateArr(e, 2, 'feedback')
                }}
              />
            </div>
            <div className="row">
              <label htmlFor="true-false-answer">Answer</label>
              <input
                type="text"
                id="true-false-answer"
                value={this.state.currentQuestion.answers[3].answer}
                onChange={(e: any) => {
                  this.updateArr(e, 3, 'answer')
                }}
              />
              <label htmlFor="true-false-percent-points">Percent Points</label>
              <input
                type="text"
                id="true-false-percent-points"
                value={this.state.currentQuestion.answers[3].percentPoints}
                onChange={(e: any) => {
                  this.updateArr(e, 3, 'percentPoints')
                }}
              />
              <label htmlFor="true-false-feed-back">feed Back</label>
              <input
                type="text"
                id="true-false-feed-back"
                value={this.state.currentQuestion.answers[3].feedback}
                onChange={(e: any) => {
                  this.updateArr(e, 3, 'feedback')
                }}
              />
            </div>
          </form>
        )
    }
  }

  render() {
    return (
      <div>
        <form>
          <label htmlFor="title">Title</label>
          <input type="text" id="title" onChange={this.updateTitle} />
          <select
            name=""
            id="question-options-dropdown"
            onChange={this.updateFormat}
          >
            <option value="true-false">true false</option>
            <option value="multiple-choice">multiple choice</option>
            <option value="best-choice">one right answer</option>
          </select>
          {this.createTable()}
          <button
            type="button"
            onClick={(e: any) => {
              // this.props.startcreateNewQuestion(this.state.currentQuestion)
            }}
          >
            Submit Question
          </button>
        </form>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  username: state.auth.username,
  email: state.auth.email,
  firstname: state.auth.firstname
})

export default connect(
  undefined,
  { startcreateNewQuestion }
)(AddQuestion)
