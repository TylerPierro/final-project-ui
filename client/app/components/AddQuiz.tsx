import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import fontawesome from '@fortawesome/fontawesome'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { startCreateNewQuestion } from '../actions/create'

interface IProps {
  startCreateNewQuestion: (any) => any
}

export class AddQuestion extends Component<IProps> {
  state = {
    quiz: {
      author: '',
      title: '',
      description: '',
      isPrivate: true
    }
  }

  private updateQuiz(e: any, arg1: string) {
    switch (arg1) {
      case 'title':
        this.setState({
          quiz: {
            ...this.state.quiz,
            title: e.target.value
          }
        })
        break
      case 'description':
        this.setState({
          quiz: {
            ...this.state.quiz,
            description: e.target.value
          }
        })
      default:
        break
    }
  }
  render() {
    return (
      <div>
        <form>
          <label htmlFor="add-quiz-title">Title</label>
          <input
            type="text"
            id="add-quiz-title"
            value={this.state.quiz.title}
            onChange={(e: any) => {
              this.updateQuiz(e, 'title')
            }}
          />
          <label htmlFor="add-quiz-description">Description</label>
          <input
            type="text"
            id="add-quiz-description"
            value={this.state.quiz.description}
            onChange={(e: any) => {
              this.updateQuiz(e, 'description')
            }}
          />
        </form>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  username: state.auth.username,
  email: state.auth.email,
  firstname: state.auth.firstname
})

export default connect<any, IProps>(
  undefined,
  { startCreateNewQuestion }
)(AddQuestion)
