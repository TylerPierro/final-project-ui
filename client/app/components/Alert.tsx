import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { Alert } from 'reactstrap';



export class AlertComponent extends React.Component<any, any> {

  constructor(props: any) {
    super(props);
  }

  public render() {
    return (
      <div>
        <Alert color="dark">
           Password updated
        </Alert>
      </div>
    );
  }
}