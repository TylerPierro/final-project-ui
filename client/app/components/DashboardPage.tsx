import React, { Component } from 'react'
import moment from 'moment'
import numeral from 'numeral'

export const DashboardPage = props => (
  <div className="dashboard-page">
    <h1>dashboard</h1>
  </div>
)

export default DashboardPage
