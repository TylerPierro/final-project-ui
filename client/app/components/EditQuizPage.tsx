import React, { Component } from 'react'
import moment from 'moment'
import numeral from 'numeral'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import Spinner from 'react-spinkit'
import Dropzone from 'react-dropzone'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  startDeleteJunction,
  startDeleteQuestion,
  startEditQuestion
} from '../actions/questions'
import { editStoreQuiz, startUpdateQuestionsDisplay } from '../actions/quizzes'

interface IProps {
  username: any
  quiz: any
  quizzes: any[]
  editStoreQuiz: (any) => any
  startEditQuestion: (any) => any
  startDeleteQuestion: (author: string, title: any) => any
  startDeleteJunction: (quizUUID: string, questionUUID: string) => any
}

export class EditQuizPage extends Component<IProps> {
  state = {
    page: 1,
    clickedQuestion: {
      author: '',
      title: '',
      uuid: '',
      format: '',
      answers: [
        {
          answer: '',
          percentPoints: 0,
          feedback: ''
        }
      ]
    },
    questionNumber: 0,
    updatedQuestions: []
  }

  params = window.location.href.split('/')
  quizUUID = this.params[4]

  page1 = () => this.setState({ page: 1 })

  page2 = () => this.setState({ page: 2 })

  private updateArr = (e: any, arg1: number, arg2: string) => {
    let newAnswersArr = this.state.clickedQuestion.answers
    newAnswersArr[arg1][arg2] = e.target.value
    this.setState({
      currentQuestion: {
        ...this.state.clickedQuestion,
        answers: newAnswersArr
      }
    })
    console.log(this.state)
  }

  private updateQuiz = (e: any) => {
    for (let item of this.state.updatedQuestions)
      this.props.startEditQuestion(item)
  }

  private saveChangeToState = (e: any) => {
    let newQArr = this.state.updatedQuestions
    newQArr.push(this.state.clickedQuestion)
    this.setState({
      ...this.state,
      updatedQuestions: newQArr
    })
    console.log(this.state.updatedQuestions)
  }

  private deletedTitle = (e: any) => {
    let title = this.state.clickedQuestion.title
    title + '--This Question has been deleted'
    this.setState({
      clickedQuestion: {
        ...this.state.clickedQuestion,
        title: title
      }
    })
  }

  private deleteQuestion = (e: any) => {
    // Something that asks them if they really want to delete it
    this.updateStore()
    this.props.startDeleteQuestion(
      this.state.clickedQuestion.author,
      this.state.clickedQuestion.title
    )
    console.log(
      this.state.clickedQuestion.author,
      this.state.clickedQuestion.title
    )
  }

  private updateStore = () => {
    let modQuiz = this.props.quiz
    for (let i = 0; i < modQuiz.questions.length; i++) {
      if ((modQuiz.questions[i].uuid = this.state.clickedQuestion.uuid)) {
        modQuiz.questions.splice(i, 1)
        //here is where we delete a question
      }
    }
    console.log('without deleted question: ', modQuiz)
    let quizList = this.props.quizzes
    for (let i = 0; i < quizList.length; i++) {
      if ((quizList[i].uuid = this.props.quiz.uuid)) {
        quizList.splice(i, 1, modQuiz)
      }
    }
    console.log('Changed:', quizList)
    this.props.editStoreQuiz(quizList)
  }

  public showQuizQuestion = (question: any, count: number, e: any) => {
    e.preventDefault()
    this.setState({
      ...this.state,
      questionNumber: count,
      clickedQuestion: question
    })
  }

  private photoUpload: HTMLInputElement

  fileSelectedHandler = e => this.setState({ selectedFile: e.target.files[0] })

  onDrop = (files: any) => {
    // get most recent file
    const file = files[0]

    // build url to s3 bucket
    // const profileUrl =
    //   'http://vocab-app-pics.s3.amazonaws.com/' +
    //   this.props.username +
    //   '/' +
    //   file.name

    this.setState({
      file
      // url: profileUrl
    })
  }

  // @ts-ignore
  render = () => {
    const { quiz } = this.props
    const { page, clickedQuestion, questionNumber } = this.state
    let count = 0
    return (
      <div className="edit-quiz-page">
        {quiz.tags === undefined && (
          <Spinner className="loading-indicator" name="ball-spin-fade-loader" />
        )}
        {quiz.tags !== undefined && (
          <main>
            <div className="quiz-container">
              <h1 className="title">{quiz.title}</h1>
              <div className="close">
                {/* <div className="close" onClick={this.onClose}> */}
                <FontAwesomeIcon icon="trash" />
                <p className="hint">Permanently delete this quiz</p>
              </div>
              <div className="tags">
                {quiz.tags.length === 0 && <p className="tag">No tags</p>}
                {quiz.tags.length > 0 &&
                  quiz.tags.map(tag => (
                    <p key={tag.allLowerCase} className="tag">
                      {tag.allLowerCase}
                    </p>
                  ))}
              </div>
              <p className="add-tag">+ tag</p>
              <div className="questions">
                {quiz.questions.map(tag => (
                  <div key={tag.allLowerCase}>
                    <p
                      className="question"
                      onClick={this.showQuizQuestion.bind(
                        this,
                        quiz.questions[count],
                        count + 1
                      )}
                    >
                      Question {(count += 1)}
                    </p>
                  </div>
                ))}
              </div>
              <p className="add-question">+ question</p>
              <button
                // onClick={this.saveChangeToState}
                className="save-quiz"
              >
                Save Quiz
              </button>

              {/* <Link
                to={`/take-quiz/${quiz.uuid}`}
                className="unset-anchor nav-link"
              >
                <div style={quizButton}>Test Quiz</div>
              </Link> */}
            </div>

            {/* {console.log(clickedQuestion)} */}
            {clickedQuestion.uuid && (
              <div key={clickedQuestion.uuid} className="question-container">
                <p className="title">Edit question</p>
                <div className="close" onClick={this.deleteQuestion}>
                  <FontAwesomeIcon icon="trash" />
                  <p className="hint">Permanently delete this question</p>
                </div>
                {page === 1 && (
                  <form className="details">
                    <div className="group">
                      <label>Question {questionNumber}</label>
                      <input placeholder={clickedQuestion.title} />
                    </div>
                    <div className="group">
                      <label>Tags</label>
                      <input placeholder="Assign tags to this question" />
                    </div>
                    <div
                      className="group photo-container"
                      onClick={() => this.photoUpload.click()}
                    >
                      <div className="group">
                        <h2 className="label">Image</h2>
                        <Dropzone onDrop={this.onDrop} className="dropzone">
                          <p className="button">+ Upload</p>
                        </Dropzone>
                        <input
                          className="file-upload"
                          style={{ display: 'none' }}
                          name="file"
                          type="file"
                          onChange={this.fileSelectedHandler}
                          ref={photoUpload => (this.photoUpload = photoUpload)}
                        />
                      </div>
                    </div>
                  </form>
                )}
                {page === 2 && (
                  <form className="options">
                    {clickedQuestion.answers.map((ans, index) => (
                      <div key={ans.answer}>
                        <div className="group">
                          <label htmlFor="true-false-answer" className="label">
                            Choice
                          </label>
                          <textarea
                            id="true-false-answer"
                            value={ans.answer}
                            className="input"
                            onChange={(e: any) => {
                              this.updateArr(e, index, 'answer')
                            }}
                            data-enable-grammarly="false"
                          />
                        </div>
                        <div className="group">
                          <label
                            htmlFor="true-false-percent-points"
                            className="input"
                          >
                            Percent Points
                          </label>
                          <textarea
                            id="true-false-percent-points"
                            value={ans.percentPoints}
                            onChange={(e: any) => {
                              this.updateArr(e, index, 'percentPoints')
                            }}
                            data-enable-grammarly="false"
                          />
                        </div>
                        <div className="group">
                          <label htmlFor="true-false-feed-back">Feedback</label>
                          <textarea
                            id="true-false-feed-back"
                            value={ans.feedback}
                            className="input"
                            onChange={(e: any) => {
                              this.updateArr(e, index, 'feedback')
                            }}
                            data-enable-grammarly="false"
                          />
                        </div>
                      </div>
                    ))}
                  </form>
                )}
                <div className="page-navigation">
                  <p
                    className={page === 1 ? 'page-1 active' : 'page-1'}
                    onClick={this.page1}
                  >
                    1
                  </p>
                  <p
                    className={page === 2 ? 'page-2 active' : 'page-2'}
                    onClick={this.page2}
                  >
                    2
                  </p>
                </div>
                {page === 2 && (
                  <button
                    onClick={this.saveChangeToState}
                    className="save-question"
                  >
                    Save
                  </button>
                )}
              </div>
            )}
          </main>
        )}
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  username: state.auth.username,
  quiz:
    state.quizzes.all !== undefined &&
    state.quizzes.all.find(quiz => quiz.uuid === props.match.params.uuid),
  quizzes: state.quizzes.all
  // clickedQuestion: state.quizzes.clickedQuestion
})

export default connect(
  mapStateToProps,
  { editStoreQuiz, startDeleteJunction, startDeleteQuestion, startEditQuestion }
)(EditQuizPage)
