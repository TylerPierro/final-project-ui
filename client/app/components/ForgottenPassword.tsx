import React, { Component } from 'react'
import { Link } from 'react-router-dom'

interface IState {
  
}

export class ResetPasswordPage extends Component<null, IState> {
  state = {

  }

// public forgotPassword = () => {
//     CognitoUser.forgotPassword({
//         onSuccess: result => {
//             console.log('call result: ' + result);
//         },
//         onFailure: err => {
//             alert(err);
//         },
//         inputVerificationCode() {
//             let verificationCode = prompt('Please input verification code ' ,'');
//             let newPassword = prompt('Enter new password ' ,'');
//             cognitoUser.confirmPassword(verificationCode, newPassword, this);
//         }
//     });
// }

}