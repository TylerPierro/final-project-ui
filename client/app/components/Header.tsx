import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

// import SearchResults from './SearchResults'
import { startLogout } from '../actions/auth'

interface RProps {
  isAuthenticated: Boolean
  username: string
  photo?: string
}

interface RState {
  startLogout: () => void
}

type Props = RProps & RState

export class Header extends Component<Props> {
  state = {
    dropdownOpen: false
  }

  toggleDropdown = () =>
    this.setState({ dropdownOpen: !this.state.dropdownOpen })

  // @ts-ignore
  componentDidMount = () =>
    document.addEventListener('click', this.determineSelection, false)

  //@ts-ignore
  componentWillUnmount = () =>
    document.removeEventListener('click', this.determineSelection)

  determineSelection = ({ target }) => {
    // if you click outside, some code runs
    if (!target.closest('.section') && !target.dataset.search) this.searchBlur()

    // else if you click inside, some other code runs
  }

  searchFocus = () => {
    this.setState({ showSearch: true })
  }

  searchBlur = () => {
    this.setState({ showSearch: false })
  }

  onFieldChange = e => {
    let value = e.target.value
    this.setState({
      searchTerm: value
    } as any)
  }

  handleSubmit = e => {
    e.preventDefault()
  }

  render() {
    const { isAuthenticated, username, startLogout, photo } = this.props

    return (
      <header className="nav-header">
        {/* <SearchResults term={searchTerm} inputState={this.state.showSearch} /> */}
        <FontAwesomeIcon icon="bars" className="bars" />
        <div className="search-group">
          <FontAwesomeIcon icon="search" className="icon" />
          <form className="submit" onSubmit={this.handleSubmit}>
            <input
              type="text"
              className="input spawnSearch"
              placeholder="Search"
              // value={searchTerm}
              data-search={true}
              onChange={this.onFieldChange}
              onFocus={this.searchFocus}
            />
          </form>
        </div>

        <FontAwesomeIcon icon="bell" className="alerts" />

        <Dropdown
          isOpen={this.state.dropdownOpen}
          toggle={this.toggleDropdown}
          className="dropdown-root"
        >
          <DropdownToggle className="dropdown-toggle">
            <div
              className="image"
              style={{
                background: `url(${photo}) center / cover no-repeat`
              }}
            />
            <p className="nav-username">{username}</p>
            <FontAwesomeIcon icon="angle-down" className="icon fa-angle-down" />
          </DropdownToggle>
          <DropdownMenu
            right
            className="dropdown-menu"
            style={{
              display: this.state.dropdownOpen === false ? 'none' : 'block'
            }}
          >
            <DropdownItem className="dropdown-item">
              <Link to="/profile" className="profile">
                Profile
              </Link>
            </DropdownItem>
            <DropdownItem
              className="dropdown-item"
              onClick={() => startLogout()}
            >
              {isAuthenticated ? <p className="logout">Logout</p> : null}
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </header>
    )
  }
}

const mapStateToProps = (state): RProps => {
  return {
    isAuthenticated: !!state.auth.token,
    username: state.auth.username
    // photo: state.auth.profileImage
  }
}

export default connect<RProps, RState>(
  mapStateToProps,
  { startLogout }
)(Header)
