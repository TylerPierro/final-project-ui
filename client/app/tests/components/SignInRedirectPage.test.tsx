import React, { Component } from 'react'
import { shallow } from 'enzyme'
import { SignInRedirectPage } from '../../components/SignInRedirectPage'

/**
 * @jest-environment jsdom
 */

test('should render SignInRedirectPage component', () => {
  window.location.assign = jest.fn()
  const wrapper = shallow(<SignInRedirectPage />)

  expect(wrapper).toMatchSnapshot()
})
