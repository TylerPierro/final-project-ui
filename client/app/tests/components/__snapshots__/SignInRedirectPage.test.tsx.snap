// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignInRedirectPage component 1`] = `<div />`;
